### PR TITLE
Rails 6.0 compatibility

### PR DESCRIPTION
**Pull request by zhuravel**
Tests pass with the updated `phenix` code from https://github.com/zendesk/phenix/pull/10.

### CODE REVIEW

**Comment by bquorning** [comment]
I have configured CircleCI to allow forked PRs to run. And I’ve released Phenix v0.6.0.

If you push this branch again, I think (hope) the build is going to run.

File: test/migrator_test.rb [hunk 1]
```diff
@@ -117,15 +117,15 @@
 
   describe "#shard_status" do
     it "shows nothing if everything is ok" do
-      ActiveRecord::Migrator.shard_status([1]).must_equal([{}, {}])
+      _(ActiveRecord::Migrator.shard_status([1])).must_equal([{}, {}])
```
Comment (bquorning) [inline test/migrator_test.rb]: Can you explain this change in words, please?

Comment (zhuravel) [reply test/migrator_test.rb]: Minitest 6 introduced some deprecations: https://github.com/seattlerb/minitest/blob/96ba1dac6141599cce47ec298e7e72dc7d49812d/lib/minitest/spec.rb#L13

Comment (grosser) [reply test/migrator_test.rb]: FYI `require 'maxitest/global_must'` will silence this too, but converting is fine too 👍

**Comment by bquorning** [comment]
@zhuravel Did you look into the new connection switching API in ActiveRecord v6.0? https://github.com/rails/rails/pull/34052 and https://github.com/rails/rails/pull/35073

I must admit that I haven’t yet looked at it in details, but I imagined that it might be used to simplify how this gem works. I am actually surprised to see that it apparently still works _without_ using the new connection API.

**Comment by zhuravel** [comment]
@bquorning I did not look into how the new connection switching API can be used to simplify the gem. My goal was to upgrade our project to Rails 6 and this PR is mostly for people whose Rails 6 upgrade is also blocked by `active_record_shards`.

**Comment by bquorning** [comment]
Could you please also lock to `"phenix", ">= 0.6.0" in the gemspec?

**Comment by bquorning** [comment]
> Could you please also lock to `"phenix", ">= 0.6.0" in the gemspec?

Never mind – I’ll merge this PR first, then fix up a few things before releasing the new version.

Thanks for the contribution – and your patience with me 😄

**Comment by bquorning** [comment]
@zhuravel I have just released v3.16.0 which includes these changes. Thank you again.

**Pull request merged**